napari/plugins/_tests/test_npe2.py

```diff
@@ -42,6 +42,12 @@ def test_read(mock_pm: 'TestPluginManager'):
     assert _npe2.read(["some.randomext"], stack=True) is None
     mock_pm.commands.get.assert_not_called()
 
+    mock_pm.commands.get.reset_mock()
+    assert (
+        _npe2.read(["some.randomext"], stack=True, plugin=PLUGIN_NAME) is None
```

Comment:
Here we expect the `ValueError` to be raised, because the user chose an `npe2` plugin but it didn't work. We should also add an additional test for a non npe2 plugin so:

```python
    mock_pm.commands.get.reset_mock()
    assert (
        _npe2.read(["some.randomext"], stack=True, plugin='not-npe2-plugin') is None
    )
    mock_pm.commands.get.assert_not_called()

    with pytest.raises(ValueError, match=f"Plugin '{PLUGIN_NAME}' was selected"):
        _npe2.read(["some.randomext"], stack=True, plugin=PLUGIN_NAME)
```

Reply:
I've added these tests and they're passing, thanks :)